src/basic/scripting-literal-and-variable.typ

```diff
@@ -1007,15 +1058,41 @@ Typst的多文件特性从何而来？
 
 #exercise[
   编写函数，使用`table`（表格）元素打印任意$N times M$矩阵，例如：
-  
+
   ```typ
   #matrix-fmt(
     (1, 2, 3),
     (4, 5, 6),
     (7, 8, 9),
   )
   ```
-  
+
+  #rect(width: 100%, eval(q1.text, mode: "markup"))
+][
+  #q1
+]
+
+#let q1 = ```typ
+#let x = state("x", 0)
+#let add_x(y) = context {
+  let res = x.get() + y
+  x.update(res)
+  [x的值现在为：#res\ ]
+}
+#x.update(0)
+#add_x(3.14)
+#add_x(1.59)
+#add_x(4.01)
+```
+
+#exercise[
+  编写程序，实现如下的“加法计算器”：
```

Comment:
context还能这样用啊。不过这样玩state感觉有点不太对劲..? 我不确定

Reply:
> The [get](https://typst.app/docs/reference/introspection/state/#definitions-get) function retrieves the current value of the state. Because the value can vary over the course of the document, it is a contextual function that can only be used when [context](https://typst.app/docs/reference/context/) is available.

官网[这里](https://typst.app/docs/reference/introspection/state/#state-in-typst)给的是这么个玩意（）